src/mad_spark_alt/core/intent_detector.py

```
@@ -0,0 +1,219 @@
+            for phrase in patterns["phrases"]:
+                if phrase in question_lower:
+                    score += 2.0
+                    matched_keywords.append(phrase)
+            
+            # Normalize score - use a more reasonable normalization
+            # Don't expect all keywords to match, just a few good ones
+            if score > 0:
+                # Normalize based on actual matches, not total possible
+                normalized_score = min(score / 5.0, 1.0)  # 5 matches = 100% confidence
+            else:
+                normalized_score = 0.0
```

Ruff (0.12.2)

Use ternary operator `normalized_score = min(score / 5.0, 1.0) if score > 0 else 0.0` instead of `if`-`else`-block

(SIM108)